src/apps/insights/utils/supabaseAdapter.ts

```
@@ -0,0 +1,124 @@
+                let data = result.data || [];
+
+                if (filter) {
+                  data = data.filter(filter);
+                }
+
+                sortByColumn(data, column, options);
+                resolve({ data, error: null });
+              } catch (error: any) {
+                reject({ data: null, error });
+              }
+            });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)